### PR TITLE
Add smoke tests for `Bucket` resource

### PR DESCRIPTION
**Pull request by A-Hilaly**
Issue https://github.com/aws-controllers-k8s/community/issues/799

Description of changes:
- Add s3 `Bucket` e2e tests for create and delete operations

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jaypipes — APPROVED** [review]
++

File: test/e2e/tests/test_bucket.py [hunk 1]
```diff
@@ -1,6 +1,98 @@
...
+        # Delete k8s resource
+        _, deleted = k8s.delete_custom_resource(ref)
+        assert deleted is True
+
+        time.sleep(DELETE_WAIT_AFTER_SECONDS)
+
+        # Check S3 Bucket doesn't exists
+        exists = self.bucket_exists(s3_client, resource_name)
+        assert not exists
+
```
[the remark this answers is not in the document]
Comment (A-Hilaly) [reply test/e2e/tests/test_bucket.py]: Yup! Fixed

**Pull request merged**